pkg/api/unversioned/group_version.go

```diff
@@ -241,6 +253,28 @@ func (gv *GroupVersion) UnmarshalText(value []byte) error {
 	return gv.unmarshal(value)
 }
 
+// GroupVersions can be used to represent a set of desired group versions.
+type GroupVersions []GroupVersion
```

Comment:
Huh, that change makes this type significantly harder.  Reminding us of the example: Job.__internal.batch, ScheduledJob.__internal.batch .  Job should serialize to v1.batch, ScheduledJob should go to v2alpha1.batch.